packages/frontend/@n8n/design-system/src/v2/components/MenuItem/component-menu-item.md

```diff
@@ -0,0 +1,101 @@
+# Component specification
+
+Primitive style wrapper for re-use across different types of menus: Sidebar, DropdownMenu, ContextMenu etc. As it wraps a primitive it can be used as different types of elements a, button or other Reka components etc.
+
+- **Component Name:** N8nMenuItem
+- **Figma Component:** [Figma](https://www.figma.com/design/8zib7Trf2D2CHYXrEGPHkg/n8n-Design-System-V3?m=auto&node-id=2388-8034&t=lj9ZucXrX3UZMu4U-1)
+- **Reka UI Component:** [Primitive](https://reka-ui.com/docs/utilities/primitive)
+
+
+## Public API definition
+
+**Props**
+
+- `item: IMenuItem` Existing type from design system.
+- `collapsed?: boolean` True for collapsed sidebar where only icon is visible
```

Comment:
Nit: Since this component will not only be used in sidebar, maybe we can remove sidebar reference from this sentence.